xdsl/transforms/convert_onnx_to_linalg.py

```diff
@@ -62,24 +62,21 @@ def match_and_rewrite(self, add: onnx.Add, rewriter: PatternRewriter, /):
 class ReluOpLowering(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, relu: onnx.Relu, rewriter: PatternRewriter, /):
-        body = Region(Block(arg_types=(f64, f64)))
-        affine_map = AffineMapAttr(AffineMap.from_callable(lambda d0, d1: (d0, d1)))
+        operand = relu.operand.type
+        assert isinstance(operand, TensorType)
+        operand_rank = len(operand.get_shape())
+        body = Region(Block(arg_types=(operand.element_type, operand.element_type)))
```

Comment:
need a cast, unfortunately

Reply:
its still being annoying with the cast rn

Reply:
it doesnt like this:

` zero := arith.Constant(FloatAttr(0, operand_type.element_type)),`

Reply:
FloatAttr requires the parameter to be a specific type, `operand_type.element_type` needs to be cast to `FloatAttr[Any]`

Reply:
i tried that stuff before, it doesnt fix it

Reply:
can you push what you tried?

Reply:
done